Mathlib/Data/Real/CauSeq.lean

```diff
@@ -0,0 +1,1011 @@
+#align exists_forall_ge_and exists_forall_ge_and
+
+section
+
+variable [LinearOrderedField α] [Ring β] (abv : β → α) [IsAbsoluteValue abv]
+
+theorem rat_add_continuous_lemma {ε : α} (ε0 : 0 < ε) :
+    ∃ δ > 0, ∀ {a₁ a₂ b₁ b₂ : β}, abv (a₁ - b₁) < δ → abv (a₂ - b₂) < δ →
+      abv (a₁ + a₂ - (b₁ + b₂)) < ε :=
+  ⟨ε / 2, half_pos ε0, @fun a₁ a₂ b₁ b₂ h₁ h₂ => by
+    simpa [add_halves, sub_eq_add_neg, add_comm, add_left_comm, add_assoc] using
+      lt_of_le_of_lt (abv_add abv _ _) (add_lt_add h₁ h₂)⟩
+#align rat_add_continuous_lemma rat_add_continuous_lemma
+
+theorem rat_mul_continuous_lemma {ε K₁ K₂ : α} (ε0 : 0 < ε) :
+    ∃ δ > 0, ∀ {a₁ a₂ b₁ b₂ : β}, abv a₁ < K₁ → abv b₂ < K₂ → abv (a₁ - b₁) < δ →
+      abv (a₂ - b₂) < δ → abv (a₁ * a₂ - b₁ * b₂) < ε := by
+  have K0 : (0 : α) < max 1 (max K₁ K₂) := lt_of_lt_of_le zero_lt_one (le_max_left _ _)
+  have εK := div_pos (half_pos ε0) K0
+  refine' ⟨_, εK, @fun a₁ a₂ b₁ b₂ ha₁ hb₂ h₁ h₂ => _⟩
+  replace ha₁ := lt_of_lt_of_le ha₁ (le_trans (le_max_left _ K₂) (le_max_right 1 _))
+  replace hb₂ := lt_of_lt_of_le hb₂ (le_trans (le_max_right K₁ _) (le_max_right 1 _))
+  have :=
+    add_lt_add (mul_lt_mul' (le_of_lt h₁) hb₂ (abv_nonneg abv _) εK)
+      (mul_lt_mul' (le_of_lt h₂) ha₁ (abv_nonneg abv _) εK)
+  rw [← abv_mul abv, mul_comm, div_mul_cancel _ (ne_of_gt K0), ← abv_mul abv, add_halves] at this
+  simpa [sub_eq_add_neg, mul_add, add_mul, add_left_comm] using
+    lt_of_le_of_lt (abv_add abv _ _) this
+#align rat_mul_continuous_lemma rat_mul_continuous_lemma
+
+theorem rat_inv_continuous_lemma {β : Type _} [DivisionRing β] (abv : β → α) [IsAbsoluteValue abv]
+    {ε K : α} (ε0 : 0 < ε) (K0 : 0 < K) :
+    ∃ δ > 0, ∀ {a b : β}, K ≤ abv a → K ≤ abv b → abv (a - b) < δ → abv (a⁻¹ - b⁻¹) < ε := by
+  refine' ⟨K * ε * K, mul_pos (mul_pos K0 ε0) K0, @fun a b ha hb h => _⟩
+  have a0 := K0.trans_le ha
+  have b0 := K0.trans_le hb
+  rw [inv_sub_inv' ((abv_pos abv).1 a0) ((abv_pos abv).1 b0), abv_mul abv, abv_mul abv, abv_inv abv,
+    abv_inv abv, abv_sub abv]
+  refine' lt_of_mul_lt_mul_left (lt_of_mul_lt_mul_right _ b0.le) a0.le
+  rw [mul_assoc, inv_mul_cancel_right₀ b0.ne', ← mul_assoc, mul_inv_cancel a0.ne', one_mul]
+  refine' h.trans_le _
+  exact mul_le_mul (mul_le_mul ha le_rfl ε0.le a0.le) hb K0.le (mul_nonneg a0.le ε0.le)
+#align rat_inv_continuous_lemma rat_inv_continuous_lemma
+
+end
+
+/-- A sequence is Cauchy if the distance between its entries tends to zero. -/
+def IsCauSeq {α : Type _} [LinearOrderedField α] {β : Type _} [Ring β] (abv : β → α) (f : ℕ → β) :
+    Prop :=
+  ∀ ε > 0, ∃ i, ∀ j ≥ i, abv (f j - f i) < ε
+#align is_cau_seq IsCauSeq
+
+namespace IsCauSeq
+
+variable [LinearOrderedField α] [Ring β] {abv : β → α} [IsAbsoluteValue abv] {f g : ℕ → β}
+
+-- see Note [nolint_ge]
+--@[nolint ge_or_gt] -- Porting note: restore attribute
+theorem cauchy₂ (hf : IsCauSeq abv f) {ε : α} (ε0 : 0 < ε) :
+    ∃ i, ∀ (j) (_ : j ≥ i) (k) (_ : k ≥ i), abv (f j - f k) < ε := by
```

Comment:
@digama0, is the right thing to do with the
```
/- ./././Mathport/Syntax/Translate/Basic.lean:632:2: warning: expanding binder collection (j k «expr ≥ » i) -/
```
to just [remove the comment](https://github.com/leanprover-community/mathlib4/compare/154f76b355130e6178b0322a7b682eff1f9e0cba..mcdoll/data_real_causeq#diff-18f1434dcab7d38fad2c5144ab5f653c1eef55b9d1d49d20951e556af4ad0317L108)? Or should it have been left in place?

Reply:
I think you can use the bunder collection syntax here, i.e. `∃ i, ∀ j ≥ i,  ∀ k ≥ i, abv (f j - f k) < ε`